cmd/jaeger/internal/integration/e2e_integration.go

```diff
@@ -144,12 +126,67 @@ func (s *E2EStorageIntegration) e2eInitialize(t *testing.T, storage string) {
 	s.SpanReader, err = createSpanReader(logger, ports.QueryGRPC)
 	require.NoError(t, err)
 
+	// Wait for the binary to start and become ready to serve requests.
+	require.Eventually(t, func() bool { return s.doHealthCheck(t) },
+		60*time.Second, 3*time.Second, "%s did not start", s.BinaryName)
+	t.Logf("%s is ready", s.BinaryName)
```

Comment:
Why did you move it here? It is again breaking the logical flow and the cleanup order.

Reply:
Yeah, moving it back. Was just testing.